### PR TITLE
Cleaned up web modules & added output descriptions

### PR DESCRIPTION
**Pull request by MrMCake**
# Change

- Cleaned up network modules
  - Convert json(null) into null
  - Removed excess brackets
- added output descriptions

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update (Wiki)

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings
- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (readme)
- [x] I did format my code

### CODE REVIEW

**Comment by MariusStorhaug** [comment]
This PR is progress towards closing #410 and #238, but doesnt alone close it. So I am unlinking issues.

File: arm/Microsoft.Web/hostingEnvironments/deploy.bicep [hunk 1]
```diff
@@ -161,19 +161,19 @@ resource appServiceEnvironment_lock 'Microsoft.Authorization/locks@2016-09-01' =
...
   properties: {
-    storageAccountId: (empty(diagnosticStorageAccountId) ? json('null') : diagnosticStorageAccountId)
-    workspaceId: (empty(workspaceId) ? json('null') : workspaceId)
-    eventHubAuthorizationRuleId: (empty(eventHubAuthorizationRuleId) ? json('null') : eventHubAuthorizationRuleId)
-    eventHubName: (empty(eventHubName) ? json('null') : eventHubName)
-    logs: ((empty(diagnosticStorageAccountId) && empty(workspaceId) && empty(eventHubAuthorizationRuleId) && empty(eventHubName)) ? json('null') : diagnosticsLogs)
+    storageAccountId: empty(diagnosticStorageAccountId) ? null : diagnosticStorageAccountId
+    workspaceId: empty(workspaceId) ? null : workspaceId
+    eventHubAuthorizationRuleId: empty(eventHubAuthorizationRuleId) ? null : eventHubAuthorizationRuleId
+    eventHubName: empty(eventHubName) ? null : eventHubName
```
Comment (MariusStorhaug) [inline arm/Microsoft.Web/hostingEnvironments/deploy.bicep:175]: It seems you are restructuring some of the conditional statements, but not all. I'm not sure I see why we are doing that to some, and not here. 

It seemed we wanted a guideline where we are trying to keep the provided parameter value in the 'true' section, while the default value go in the 'false' section of the conditional statements. 

Conditional expression:
```bicep
condition ? true-value : false-value
```

Guideline:
```bicep
condition ? parameter-value : default-value
```

I will resolve this conversation as its not a breaking change.

**Review by MariusStorhaug — DISMISSED** [review]
One comment, but not breaking nor critical. Approving

File: arm/Microsoft.Web/sites/.bicep/nested_privateEndpoint.bicep [hunk 2]
```diff
@@ -5,13 +5,13 @@ param tags object
 
 var privateEndpointResourceName = last(split(privateEndpointResourceId, '/'))
 var privateEndpoint_var = {
-  name: (contains(privateEndpointObj, 'name') ? (empty(privateEndpointObj.name) ? '${privateEndpointResourceName}-${privateEndpointObj.service}' : privateEndpointObj.name) : '${privateEndpointResourceName}-${privateEndpointObj.service}')
+  name: contains(privateEndpointObj, 'name') ? (empty(privateEndpointObj.name) ? '${privateEndpointResourceName}-${privateEndpointObj.service}' : privateEndpointObj.name) : '${privateEndpointResourceName}-${privateEndpointObj.service}'
```
Comment (MariusStorhaug) [inline arm/Microsoft.Web/sites/.bicep/nested_privateEndpoint.bicep:8]:
```suggestion
  name: contains(privateEndpointObj, 'name') ? (!empty(privateEndpointObj.name) ? privateEndpointObj.name : '${privateEndpointResourceName}-${privateEndpointObj.service}') : '${privateEndpointResourceName}-${privateEndpointObj.service}'
```

File: arm/Microsoft.Web/sites/.bicep/nested_privateEndpoint.bicep [hunk 3]
```diff
@@ -5,13 +5,13 @@ param tags object
...
   subnetResourceId: privateEndpointObj.subnetResourceId
   service: [
     privateEndpointObj.service
   ]
-  privateDnsZoneResourceIds: (contains(privateEndpointObj, 'privateDnsZoneResourceIds') ? (empty(privateEndpointObj.privateDnsZoneResourceIds) ? [] : privateEndpointObj.privateDnsZoneResourceIds) : [])
-  customDnsConfigs: (contains(privateEndpointObj, 'customDnsConfigs') ? (empty(privateEndpointObj.customDnsConfigs) ? json('null') : privateEndpointObj.customDnsConfigs) : json('null'))
+  privateDnsZoneResourceIds: contains(privateEndpointObj, 'privateDnsZoneResourceIds') ? (empty(privateEndpointObj.privateDnsZoneResourceIds) ? [] : privateEndpointObj.privateDnsZoneResourceIds) : []
+  customDnsConfigs: contains(privateEndpointObj, 'customDnsConfigs') ? (empty(privateEndpointObj.customDnsConfigs) ? null : privateEndpointObj.customDnsConfigs) : null
```
Comment (MariusStorhaug) [inline arm/Microsoft.Web/sites/.bicep/nested_privateEndpoint.bicep:14]:
```suggestion
  privateDnsZoneResourceIds: contains(privateEndpointObj, 'privateDnsZoneResourceIds') ? (!empty(privateEndpointObj.privateDnsZoneResourceIds) ? privateEndpointObj.privateDnsZoneResourceIds : []) : []
  customDnsConfigs: contains(privateEndpointObj, 'customDnsConfigs') ? (!empty(privateEndpointObj.customDnsConfigs) ? privateEndpointObj.customDnsConfigs : null) : null
```

File: arm/Microsoft.Web/serverfarms/deploy.bicep [hunk 4]
```diff
@@ -67,16 +67,16 @@ module pid_cuaId '.bicep/nested_cuaId.bicep' = if (!empty(cuaId)) {
...
   name: appServicePlanName
-  kind: ((serverOS == 'Windows') ? '' : 'linux')
+  kind: serverOS == 'Windows' ? '' : 'linux'
   location: location
   tags: tags
   sku: sku
   properties: {
     workerTierName: workerTierName
-    hostingEnvironmentProfile: (empty(appServiceEnvironmentId) ? json('null') : hostingEnvironmentProfile)
+    hostingEnvironmentProfile: empty(appServiceEnvironmentId) ? null : hostingEnvironmentProfile
```
Comment (MariusStorhaug) [inline arm/Microsoft.Web/serverfarms/deploy.bicep:76]:
```suggestion
    hostingEnvironmentProfile: !empty(appServiceEnvironmentId) ? hostingEnvironmentProfile : null
```

File: arm/Microsoft.Web/hostingEnvironments/deploy.bicep [hunk 5]
```diff
@@ -161,19 +161,19 @@ resource appServiceEnvironment_lock 'Microsoft.Authorization/locks@2016-09-01' =
...
+    logs: empty(diagnosticStorageAccountId) && empty(workspaceId) && empty(eventHubAuthorizationRuleId) && empty(eventHubName) ? null : diagnosticsLogs
```
Comment (MariusStorhaug) [inline arm/Microsoft.Web/hostingEnvironments/deploy.bicep:176]:
```suggestion
    storageAccountId: !empty(diagnosticStorageAccountId) ? diagnosticStorageAccountId : null
    workspaceId: !empty(workspaceId) ? workspaceId : null
    eventHubAuthorizationRuleId: !empty(eventHubAuthorizationRuleId) ? eventHubAuthorizationRuleId : null
    eventHubName: !empty(eventHubName) ? eventHubName : null
    logs: !empty(diagnosticStorageAccountId) || !empty(workspaceId) || !empty(eventHubAuthorizationRuleId) || !empty(eventHubName) ? diagnosticsLogs : null
```

File: arm/Microsoft.Web/sites/.bicep/nested_privateEndpoint.bicep (hunk 2 above)
Comment (MrMCake) [reply arm/Microsoft.Web/sites/.bicep/nested_privateEndpoint.bicep]: Updated

**Review by MariusStorhaug — APPROVED** [review]
Conversations resolved. Approved.

**Pull request merged**